CATesting/config.yaml

```diff
@@ -0,0 +1,119 @@
+# Constants
+{{$PODS_PER_NODE := DefaultParam .PODS_PER_NODE 10}}
+{{$qps := DefaultParam .qps 10}}
+# Adjust the following two for your cluster -
+# kubectl get node -o json | jq -r ".items[].status.allocatable.cpu"
+{{$NODE_FREE_CPU := DefaultParam .NODE_FREE_CPU 1}}
+# kubectl get node -o json | jq -r ".items[].status.allocatable.memory"
+# In MB
+{{$NODE_FREE_MEM := DefaultParam .NODE_FREE_MEM 3840}}
+
+# Variables
+{{$podsPerNamespace := MultiplyInt $PODS_PER_NODE .Nodes}}
+{{$totalPods := MultiplyInt $PODS_PER_NODE .Nodes}}
+# Pod Resource proportional to node to ensure consistent triggering of scale-up
+{{$POD_CPU_MILLI := DivideFloat $NODE_FREE_CPU (MultiplyFloat $PODS_PER_NODE .9)}}
+{{$POD_MEMORY_MB := DivideFloat $NODE_FREE_MEM (MultiplyFloat $PODS_PER_NODE .9)}}
+
+name: Simple_Bring_Up
+namespace:
+  number: 1
+tuningSets:
+- name: Uniformqps
+  qpsLoad:
+    qps: {{$qps}}
+
+steps:
+- name: Starting Measurements
+  measurements:
+  - Identifier: ScaleUpTimer
+    Method: Timer
+    Params:
+      action: start
+      label: scale_up_time
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: start
+      labelSelector: group = nichotr
+      threshold: 1h
+  - Identifier: SchedulingThroughput
+    Method: SchedulingThroughput
+    Params:
+      action: start
+      labelSelector: group = nichotr
+
+- name: Creating pods
+  phases:
+    - namespaceRange:
+        min: 1
+        max: 1
+      replicasPerNamespace: 1
+      tuningSet: Uniformqps
+      objectBundle:
+        - basename: deployment
+          objectTemplatePath: deployment.yaml
+          templateFillMap:
+            Replicas: {{$podsPerNamespace}}
+            Group: nichotr
+            CpuRequest: {{$POD_CPU_MILLI}}m
+            MemoryRequest: {{$POD_MEMORY_MB}}M
+
+- name: Collecting pod measurements
+- measurements:
+    - Identifier: WaitingForPods
+      Method: WaitForRunningPods
+      Params:
+        labelSelector: group = nichotr
+        desiredPodCount: {{$totalPods}}
+        timeout: 1h
+
+- name: Timing Separation For Scale Up
+- measurements:
+    - Identifier: ScaleUpTimer
+      Method: Timer
+      Params:
+        action: stop
+        label: scale_up_time
+    - Identifier: ScaleDownTimer
+      Method: Timer
+      Params:
+        action: start
+        label: scale_down_time
+
+- name: Deleting pods
+  phases:
+    - namespaceRange:
+        min: 1
+        max: 1
+      replicasPerNamespace: 0
+      tuningSet: Uniformqps
+      objectBundle:
+        - basename: deployment
+          objectTemplatePath: deployment.yaml
+
+- name: Waiting for Deletion
+- measurements:
+    - Identifier: WaitingForPods
+      Method: WaitForRunningPods
+      Params:
+        labelSelector: group = nichotr
+        desiredPodCount: 0
+        timeout: 1h
+
+- name: Timing Separation for Scale Down
```

Comment:
I like how the measurements are separated between scale up and scale down. Consider

"start-scale-up-measurements"
"stop-scale-up-measurements"
"start-scale-down-measurements"
"stop-scale-down-measurements"

Reply:
Will do.